### PR TITLE
#47 complete signup

### PR DESCRIPTION
**Pull request by nazar**
References #47

### CODE REVIEW

**Comment by nazar** [comment]
also closes #43

File: api/schemas/user.js [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
-    user: (obj, { id }) => User.query().findById(id)
+    me: (obj, obj2, context) => getOrCreateUserByAuth0Id(context.req.user.sub)
+  },
+  Mutation: {
+    updateMe: (obj, { input }, context) => updateUserAndAuth0(input, context.req.user.sub),
+    validUsername: (obj, { username }, context) => {
+      return User
+        .query()
+        .count('username')
+        .where({ username })
```
Comment (mikelax) [inline api/schemas/user.js:48]: @nazar  do you think that we need to "normalize" the username
ie. lowercase, remove spaces and special chars to turn into url slug?

**Review by mikelax — APPROVED** [review]
looks really good! 
there are a couple of items we can discuss as separate tickets.

File: api/schemas/user.js (hunk 1 above)
Comment (nazar) [reply api/schemas/user.js:48]: hey @mikelax - I was thinking about that as well and was looking into validation libraries that would ease the UI/API rule checking on usernames (and names) - alpha only, min and max length etc....

In the past I've used something like [checkit](https://github.com/tgriesser/checkit). Unfortunately that library has been abandoned (along with bookshelfjs - same author). Another alternative I looked at was [validate.js](https://validatejs.org/). Another alternative is [Joi](https://github.com/hapijs/joi), but that is a rather chunky library and I'm not too fond of it.

In the end I left validation out of the PR but it is something we should revisit 👍

File: auth0/rules/update-profile-with-custom-profile.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      user.name = user.user_metadata.name;
+    }
+
+    if (user.user_metadata.username) {
+      user.nickname = user.user_metadata.username;
+    }
+  }
+  
+  callback(null, user, context);
+}
```
Comment (nazar) [inline auth0/rules/update-profile-with-custom-profile.js:17]: hey @mikelax - I didn't know the auth0 user 'name' or 'picture' could be updated. I would have done that from the API - I should have checked the docs.

**Pull request merged**